CMakeLists.txt

```diff
@@ -2,7 +2,7 @@
 # CMake Build Script for the HDF5 REST VOL connector #
 #----------------------------------------------------#
 
-cmake_minimum_required (VERSION 3.10)
+cmake_minimum_required (VERSION 3.10...3.29)
```

Comment:
I don't think we should be setting a maximum version here, as it would then need to be maintained as new versions of CMake are released. Otherwise, it could result in surprising behavior down the line.